pkg/generate/app/cmd/newapp.go

```diff
@@ -710,6 +711,23 @@ func (c *AppConfig) Run() (*AppResult, error) {
 		}
 	}
 
+	// before returning, if this is a template run, and environment variables were passed in, let's apply to the environment variables
+	// to every pod template object
+	if len(templateObjects) > 0 {
+		for i := range objects {
```

Comment:
you should iterate over templateObjects since generated objects already have the environment applied to them.